h11/_connection.py

```diff
@@ -243,15 +267,16 @@ def _process_event(self, role, event):
 
         # Then perform the updates triggered by it.
 
-        # self._request_method
-        if type(event) is Request:
+        if isinstance(event, Request):
```

Comment:
Is this an open Mypy bug? If not would be worth filing, `type(x) is y` is faster than `isinstance()`

Reply:
https://mypy.readthedocs.io/en/stable/common_issues.html#complex-type-tests

Bit of a pain - we could drop an assert in though rather than use a cast?

Reply:
Fixed, with only a couple of casts.